### PR TITLE
DestinationRule Analyzer against no caCertificates

### PR DESCRIPTION
**Pull request by nschhina**
#Ref: https://github.com/istio/istio/issues/25652

Cherrypick to 1.6 and 1.5(if analyzer exists)

edit: Analyzer exists in 1.5, so we can cherrypick to 1.5 too

### CODE REVIEW

**Comment by nschhina** [comment]
cc: @myidpt  @JimmyCYJ  @howardjohn

**Comment by JimmyCYJ** [comment]
This is great! Thanks @nschhina

**Comment by nschhina** [comment]
@therealmitchconnors this is a security vulnerability to it's important that we cherrypick this back as advised by product-security team and security wg also wants this to be backported. As for 1.7 we are still not sure how we want to approach this, we'll be discussing this in today's security wg meeting and then decide

**Comment by nschhina** [comment]
So just had the working group meeting and we've decided to cherrypick this to 1.5,1.6 and 1.7 and implement a proper fix in 1.8 release. @therealmitchconnors any particular concerns with cherrypicking this back to 1.5 and 1.6 given this is a security issue and it's desirable this information is propagated to end users

**Comment by therealmitchconnors** [comment]
@nschhina is the analyzer the only patch for this vulnerability?  We don't have any history of patching istioctl for vulnerabilities, and I don't think we have previously published guidance asking users to update istioctl when they update their control plane on a patch version.

**Comment by nschhina** [comment]
This is just to be used as medium to guide the end user regarding this "issue", there is another patch which involves using environment variables to enable fallbacks, so when that patch is cleared we can update the error to point the user to enable the fallback. This is an effort to completely disable such deployments slowly and slowly in the future as this is a breaking change.

**Comment by nschhina** [comment]
@therealmitchconnors  I wouldn't say we're patching istioctl for vulnerability, it's just something that should have existed in 1.5-1.6 anyways. But it being a breaking change we can't really enforce this during istiod validation. Will people who are using 1.5 and 1.6 start seeing this error if they run analyzer suddenly- yes. But that's the point, this already exists in API documentation but seems like nobody is reading the fine print :/

**Comment by therealmitchconnors** [comment]
I'm not worried about a new message showing up unexpectedly.  I am worried that users who patch their istio installation may not patch istioctl as well.  I think a lot of users only update istioctl when they upgrade a minor version, meaning porting this to 1.5 and 1.6 will not be visible to most users...

Also, if this change is _only_ for 1.5 and 1.6, we should make the commit directly to those branches, rather than to master, so that this doesn't unexpectedly show up in 1.8 after skipping 1.7

**Comment by nschhina** [comment]
Oh we're not skipping 1.7, I didn't explain right, For 1.5-1.7 it'll be this analyzer + environment variable to opt into defaulting which the analyzer would suggest. By 1.8 we'll probably have a better fix

**Comment by nschhina** [comment]
what do you suggest we do for those minor
 releases

**Comment by JimmyCYJ** [comment]
@therealmitchconnors we would like this to go into 1.7 and master, could you let us know how to proceed?

File: releasenotes/notes/dr-analyzer.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+apiVersion: release-notes/v2
+kind: feature
+area: networking
+releaseNotes: |
```
Comment (fpesce) [inline releasenotes/notes/dr-analyzer.yaml:4]:
```suggestion
securityNotes: |
```

Comment (fpesce) [reply releasenotes/notes/dr-analyzer.yaml:4]: What do you think of making that feature as part of security notes @brian-avery ?

Comment (nschhina) [reply releasenotes/notes/dr-analyzer.yaml:4]: Normally SecurityNotes all have a CVE associated, will it be okay as is?

Comment (fpesce) [reply releasenotes/notes/dr-analyzer.yaml:4]: I think it's okay to have non-CVE security notes to indicate changes in Istio relevant to product security.
Cc the product security WG for opinions: @brian-avery @ericvn @jacob-delgado @JimmyCYJ @duderino @knrc @linsun @mandarjog @PiotrSikora @rshriram @yangminzhu @liminw @nrjpoddar

Comment (JimmyCYJ) [reply releasenotes/notes/dr-analyzer.yaml:4]: Thanks @fpesce I am okay with a non-CVE security note.

Comment (nrjpoddar) [reply releasenotes/notes/dr-analyzer.yaml:4]: LGTM

Comment (nschhina) [reply releasenotes/notes/dr-analyzer.yaml]: I changed it to securityNotes as discussed. I added description to make upgradeNotes more clearer as well cc: @therealmitchconnors

**Comment by nrjpoddar** [comment]
We should also add an analyzer for SAN verification being skipped. Thanks for this great analyzer!

**Pull request merged**